pkg/operator/configobservation/apiserver/observe_cors.go

```diff
@@ -0,0 +1,55 @@
+package apiserver
+
+import (
+	"reflect"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+)
+
+var clusterDefaultCORSALlowedOrigins = []string{
+	`//127\.0\.0\.1(:|$)`,
+	`//localhost(:|$)`,
+}
+
+func ObserveExtraCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	const corsAllowedOriginsPath = "corsAllowedOrigins"
+
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+	currentCORSAllowedOrigins, _, err := unstructured.NestedStringSlice(existingConfig, corsAllowedOriginsPath)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentCORSAllowedOrigins) > 0 {
+		if err := unstructured.SetNestedStringSlice(prevObservedConfig, currentCORSAllowedOrigins, corsAllowedOriginsPath); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warningf("apiserver.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return prevObservedConfig, errs
+	}
+
+	newCORSAllowedOrigins := append(clusterDefaultCORSALlowedOrigins, apiServer.Spec.ExtraCORSAllowedOrigins...)
```

Comment:
clusterDefaultCORSALlowedOrigins must go to line 24.

Reply:
This should now be fixed